### ISSUE TITLE
AROS loses network connection, "no buffer space available"

### ISSUE THREAD

**Issue by subocz** [deadwood2/AROS#132]
![buffers](https://github.com/deadwood2/AROS/assets/15143958/a4b523ce-352f-4717-9d62-8e79d5542677)

1. launch Shell and type the command "ping google.com"
2. launch OWB and go to the aros archive page. Download some files.

While the files are being downloaded, AROS loses its connection to the Internet, and ping starts sending the message "no buffer space available".

This behavior occurs especially on:
- nvidianet.device (tested on ASUS M2N68-AM SE2)
- pcnet32.device (tested on VirtualBox 7.0.12)

I also manage to trigger this bug on rtl8139.device (ASUS M2N68-AM SE2 + pci network card), but not every time, easier when I download several files simultaneously and browse the web at the same time.

Despite trying, I have not been able to trigger this bug on ASUS M2N68-AM SE2 with USB internet (usbrndis.device).

**Comment by ncafferkey** [comment]
Try "ping -f" to your router to perhaps reproduce this problem more easily. E.g. ping -f 192.168.0.1

I would be good to know also if this bug definitely needs both ping and OWB or if just one of them can trigger it.

**Comment by subocz** [comment]
Using ping is not necessary to trigger the bug. Recently I discovered that ping gives the message "no buffer space available". However, the bug also occurs when I use only OWB or when I use, for example, only MPlayer with Internet radio. Today I checked "ping -f" and on nvidianet.device alone this command without other programs also triggers the bug.

**Comment by ncafferkey** [comment]
Is this a regression with a particular build (sorry, I haven't been keeping up with recent test builds)?

**Comment by subocz** [comment]
I don't think this is a regression in recent builds. This behavior occurs in the latest builds, but also in Icaros Desktop 2.3 from 2018.

![icarosdesktop](https://github.com/deadwood2/AROS/assets/15143958/99259b0f-397f-42b3-90fb-c359915c22c2)

**Comment by ncafferkey** [comment]
I wonder if this is connected to the USB problems with nVidia chipsets. Are you using any USB devices on the M2N68-AM? I tried "ping -f" on mine today with Icaros 2.3 and it ran for three hours without errors (apart from 4% packet loss, which isn't too bad for a ping flood). I don't have any USB devices connected, not even keyboard or mouse.

**Comment by subocz** [comment]
I don't have any USB devices connected. Are you pinging the router or the external domain? In my case it is much faster to trigger a bug when I ping an external domain (e.g. google.com).

**Comment by ncafferkey** [comment]
I was pinging a router. I pinged google.com just now but only for five minutes (as it's not considered polite to ping flood the internet). How long do you typically have to wait before getting errors?

**Comment by subocz** [comment]
In the case of nvidianet.device pinging google.com with -f crashes after a few minutes, and sometimes even after less than a minute

**Comment by subocz** [comment]
I recorded a video (VirtualBox, icaros desktop, pcnet32.device). Bug under two minutes.

https://drive.google.com/file/d/19kUkYJeS56FYPEX7aKFmDd0TQqYIDa1s/view?usp=sharing

**Comment by deadwood2** [comment]
I can confirm this behavior with AROS One 2.4 under VirtualBox 7.0.12. In my case it takes less then a minute with ping -f google.com to start getting "No buffer" errors.

**Comment by subocz** [comment]
In this video, I am turning on the ASUS M2N68-AM SE2. The only thing I have connected to my computer is:
- ps/2 mouse
- ps/2 keyboard
- IDE HDD
System: Latest build of AROS x86 (ABIv0 20220318-1)
Driver: nvidianet.device

https://drive.google.com/file/d/1i8izgPxhsPKSIcZOzoqw800Atb_K0etL/view?usp=sharing

**Comment by subocz** [comment]
23304 bytes. The same as in the latest buid:
https://github.com/deadwood2/AROS/releases/download/ABIv0_20220318-1/AROS-20220318-1-pc-i386-boot-iso.zip

**Comment by ncafferkey** [comment]
I retested with the version from that build but there's no difference for me. Do you have any PCI cards installed?

**Comment by subocz** [comment]
As I wrote before: only mouse and ps/2 keyboard, no other accessories or extensions

**Comment by subocz** [comment]
BIOS: 1801 (latest)

**Comment by ncafferkey** [comment]
Have you tried Icaros on that machine, or only on VirtualBox?

**Comment by subocz** [comment]
I tested icaros only on VirtualBox.

**Comment by ncafferkey** [comment]
I just realised my board probably isn't an SE2. Can you post a screengrab of the Ethernet information from PCITool please?

**Comment by deadwood2** [comment]
FYI, I tried my hardware M2N68-AM SE2 and it takes less then 10 seconds to get "No buffer" error with ping -f google.com using nvidianet.device.

Here is PCIInfo:

Driver Name: pcipc.hidd
Direct bus: yes
IÒ base: 0x00000000
Hardware info: IA32-native Direct Access PCI Bus Controller
Description: Bridge Other 
Vendor name: NVIDIA Corporation
Product name: MCP61 Ethernet
Subsystem: ASUSTeK Computer Inc. : M4N68T series motherboard
Used by: nvidianet.device
VendorID: 0x10de
ProductID: 0x03ef
RevisionID: 0xa2
Class: 0x06
Subclass: 0x80
Interface: 0x00
ROM Base: N/A
ROM Size: N/A
IRQ: 10 (A)
Status: IO: yes, MEM: yes, Master: yes, PaletteSnoop: no, 66MHz capable: yes
RangeList_0: Bar0: Memory at 0xdfffd000 (size 0x1000)
RangeList_1: Bar1: I/O at 0xe480 (size 0x8)
RangeList_2: Bar2: I/O at 0xe000 (size 0x8)
RangeList_3: Bar3: I/O at 0xdc00 (size 0x4)
RangeList_4: Bar4: I/O at 0xffa0 (size 0x10)
RangeList_5: Bar5: Memory at 0xdfffc000 (size 0x1000)

**Comment by ncafferkey** [comment]
Mine's an MCP67 with product ID 0x054c. I may look into acquiring an SE2 or another motherboard with MCP61 in order to debug it.

Overall, these errors look to be caused by individual driver bugs rather than a general networking bug. I think a lot of our network drivers behave poorly when stress tested unfortunately. We may want to open bug reports for individual drivers to replace this one.

**Comment by subocz** [comment]
![423036689_361126900151119_8595355379089324160_n](https://github.com/deadwood2/AROS/assets/15143958/96393b8c-18b1-4529-bb42-7f1a6a306b2f)
